### PR TITLE
Device segment filters

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | n
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Required: )
#### Description:
Add new filters:

- Segment filter by device type (smartphone, table, desktop....) - multiple selectbox
- Segment filter by device brand (Samsung, Apple, Acer, Tesla...) - multiple selectbox
- Segment filter by device OS  (Android, Chrome Os, iOs....) - multiple selectbox
- Segment filter by device model  (Samsung Galaxy, iPhone s6...)  - text filter

#### Steps to test this PR:
1.  Apply PR
2.  Test filters from descriptions

Fund this feature
https://mautic-fund.kuzmany.biz/campaign/3/device-segment-filter-and-campaign-condition

Thanks to sponsor of this freature reqest www.SvetZitrka.cz 

device

### CODE REVIEW

**Comment by kuzmany** [comment]
@escopecz It's ready for test. Everything looks good, my tests works.

File: app/bundles/LeadBundle/Entity/LeadListRepository.php [hunk 1]
```diff
@@ -1223,10 +1271,22 @@ public function getListFilterExpr($filters, &$parameters, QueryBuilder $q, $not
...
+                            break;
+                        case 'device_os':
+                            $table  = 'lead_devices';
+                            $column = 'device_os_name';
+                            break;
                     }
 
                     $subExpr->add(
-                        $subQb->expr()->in(sprintf('%s.%s', $alias, $column), $details['filter'])
+                        $subQb->expr()->in(sprintf('%s.%s', $alias, $column),  "'".implode("','", $details['filter'])."'")
```
Comment (escopecz) [inline app/bundles/LeadBundle/Entity/LeadListRepository.php]: The `in` method works correctly only with second param type of `array`, not `string`. (if the array contains more than 1 items)

**Comment by virlatinus** [comment]
I tested the new segments and they work.

**Comment by matishaladiwala** [comment]
works as expected. would be great to have these same filters for reports as well.

**Pull request merged**